App/Sagas/ImageSharingTriggers.ts

```diff
@@ -37,11 +38,13 @@ export function * handleImageUploadComplete (action: ActionType<typeof Processin
         }
       }
     } catch (e) {}
-    const allComplete: boolean = yield select(allUploadsComplete, processingImage.uuid)
-    const alreadySharing = processingImage.status === 'sharing' || processingImage.status === 'complete'
-    if (allComplete && !alreadySharing) {
-      yield call(shareToThread, processingImage.uuid)
-    }
+  }
+}
+
+export function * startMonitoringExistingUploads () {
```

Comment:
If the app is booted up while there are uploads in flight, this saga kicks off the monitoring of those in flight uploads so we'll know when they've completed and move on to sharing the image.